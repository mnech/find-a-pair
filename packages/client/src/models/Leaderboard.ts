/**Минимальный номер страницы для отображения*/
export const paginationMin = 1;
/**Максимальный номер страницы для отображения*/
export const paginationMax = 10;
/**Количество записей для отображения*/
export const limitPage = 10;

/**
 * Интерфейс данных с рейтингами
 *
 * name Имя пользователя
 * score Рейтинг
 */
export interface IUserScore {
  name: string;
  score: number;
}

/**
 * Название поля с рейтингом для нашей игры
 */
export const ratingFieldName = 'findAPairScore';

/**
 * Интерфейс данных с рейтингами для API
 *
 * ratingFieldName Название поля с рейтингом
 * cursor Текущая страница
 * limit Количество записей на странице
 */
export interface ILeaderboardRequest {
  ratingFieldName: string;
  cursor: number;
  limit: number;
}

/**
 * Интерфейс стейта для лидерборда
 *
 * data Данные для отображения
 * request Запрос
 */
export interface ILeaderboardState {
  data: IUserScore[];
  request: ILeaderboardRequest;
}
