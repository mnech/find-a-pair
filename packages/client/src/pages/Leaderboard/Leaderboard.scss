.container {
  padding: 10% 0;
  display: flex;
  flex-direction: column;
  align-content: center;
}
