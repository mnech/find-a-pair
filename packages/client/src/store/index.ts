import { configureStore, PreloadedState } from '@reduxjs/toolkit';
import combineReducers from '../reducers/index';

export const store = configureStore({
  reducer: combineReducers,
});

export const setupStore = (preloadedState?: PreloadedState<RootState>) => {
  return configureStore({
    reducer: combineReducers,
    preloadedState,
  });
};

export type RootState = ReturnType<typeof store.getState>;

export type AppStore = ReturnType<typeof setupStore>;
