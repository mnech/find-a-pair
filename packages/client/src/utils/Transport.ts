import axios from 'axios';
import { Methods } from '../api/consts';

export type TRequest = {
  method?: Methods
  url: string
  data?: unknown
};

export default class Transport {
  protected axios = axios;

  constructor() {
    this.axios.defaults.withCredentials = true;
  }

  public request({ method = Methods.Get, url, data }: TRequest) {
    return this.axios({ method, url, data });
  }
}
