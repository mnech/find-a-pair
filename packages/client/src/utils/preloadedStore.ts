import {
  limitPage,
  paginationMin,
  ratingFieldName,
} from '../models/Leaderboard';

export const preloadedState = {
  leaderboard: {
    //пока заглушка
    data: [
      {
        name: 'User',
        score: 1,
      },
      {
        name: 'Jane',
        score: 2,
      },
      {
        name: 'James',
        score: 7,
      },
      {
        name: 'Luis',
        score: 10,
      },
    ],
    request: {
      ratingFieldName: ratingFieldName,
      cursor: paginationMin,
      limit: limitPage,
    },
  },
};
