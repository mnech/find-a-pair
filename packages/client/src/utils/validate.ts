import { Regexp } from '../consts';

export const regexpTest = (type: keyof typeof Regexp, str: string) => {
  return Regexp[type].test(str);
};
