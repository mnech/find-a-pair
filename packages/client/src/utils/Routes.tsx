import { ErrorBoundary } from './ErrorBoundary';
import { Auth } from '../pages/Auth/Auth';
import { Leaderboard } from '../pages/Leaderboard/Leaderboard';
import Game from '../pages/Game/Game';
import Forum from '../pages/Forum/Forum';
import Profile from '../pages/Profile/Profile';
import { StartGameScreen } from '../pages/StartGameScreen/StartGameScreen';
import Signup from '../pages/SignUp/SignUp';
import { routes } from '../models/App';
import { EndGameScreen } from '../pages/EndGameScreen/EndGameScreen';
export const staticRoutes = [
  {
    path: routes.auth,
    errorElement: <ErrorBoundary />,
    children: [
      {
        index: true,
        path: routes.auth,
        element: <Auth />,
      },
      {
        path: routes.profile,
        element: <Profile />,
      },
      {
        path: routes.signUp,
        element: <Signup />,
      },
      {
        path: routes.leaderboard,
        element: <Leaderboard />,
      },
      {
        path: routes.forum,
        element: <Forum />,
      },
      {
        path: routes.game,
        element: <Game />,
      },
      {
        path: routes.startGame,
        element: <StartGameScreen />,
      },
      {
        path: routes.endGame,
        element: <EndGameScreen />,
      },
    ],
  },
];
