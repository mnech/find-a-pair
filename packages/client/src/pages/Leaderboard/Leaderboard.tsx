import Table from 'react-bootstrap/Table'
import { Pagination } from './components/Pagination'
import { useState } from 'react'
import { IUserRate, paginationMin } from '../../models/Leaderboard'
import './Leaderboard.scss'
import { ErrorBoundary } from '../../utils/ErrorBoundary'

const data: IUserRate[] = [
  {
    name: 'User',
    rate: 1,
  },
  {
    name: 'Jane',
    rate: 2,
  },
  {
    name: 'James',
    rate: 7,
  },
  {
    name: 'Luis',
    rate: 10,
  },
]
export const Leaderboard = () => {
  const [isActivPage, setIsActivPage] = useState(paginationMin)

  const renderRow = (row: IUserRate, idx: number): JSX.Element => {
    return (
      <tr key={row.name}>
        <td>{idx + 1}</td>
        <td>{row.name}</td>
        <td>{row.rate}</td>
      </tr>
    )
  }

  return (
    <ErrorBoundary>
      <div className="container-md overflow-auto mt-5 w-50 h-auto container">
        <div className="h3 pb-3">LeaderBoard</div>
        <Table
          striped
          bordered
          hover
          variant="striped"
          size="md"
          className="pb-3">
          <thead>
            <tr>
              <th>#</th>
              <th>Имя</th>
              <th>Рейтинг</th>
            </tr>
          </thead>
          <tbody>
            {data?.map((userRate, idx) => renderRow(userRate, idx))}
          </tbody>
        </Table>
        <Pagination page={isActivPage} onChange={setIsActivPage} />
      </div>
    </ErrorBoundary>
  )
}
