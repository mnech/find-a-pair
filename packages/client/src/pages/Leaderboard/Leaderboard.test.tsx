import * as renderer from 'react-test-renderer';
import { Leaderboard } from './Leaderboard'
import { renderWithProviders } from '../../utils/configureStore'
import { setupStore } from '../../store'
import { Provider } from 'react-redux'
import { getPreloadedState } from '../../utils/preloadedStore'


  test ('renders correctly', () =>{
    const tree = renderer
      .create(renderWithProviders(<Leaderboard />))
      // .create(<Provider store={setupStore(getPreloadedState())}><Leaderboard/></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();

    // const {asFragment} = render(<Leaderboard/>)
    // const firstRender = asFragment()
    // expect(firstRender).toMatchSnapshot(asFragment())

  })

