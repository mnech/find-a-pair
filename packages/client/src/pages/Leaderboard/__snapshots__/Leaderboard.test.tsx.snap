// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="container-md overflow-auto mt-5 w-50 h-auto container"
  >
    <div
      class="h3 pb-3"
    >
      LeaderBoard
    </div>
    <table
      class="pb-3 table table-striped table-md table-striped table-bordered table-hover"
    >
      <thead>
        <tr>
          <th>
            #
          </th>
          <th>
            Имя
          </th>
          <th>
            Рейтинг
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1
          </td>
          <td>
            User
          </td>
          <td>
            1
          </td>
        </tr>
        <tr>
          <td>
            2
          </td>
          <td>
            Jane
          </td>
          <td>
            2
          </td>
        </tr>
        <tr>
          <td>
            3
          </td>
          <td>
            James
          </td>
          <td>
            7
          </td>
        </tr>
        <tr>
          <td>
            4
          </td>
          <td>
            Luis
          </td>
          <td>
            10
          </td>
        </tr>
      </tbody>
    </table>
    <ul
      class="w-25 pagination pagination-sm"
    >
      <li
        class="page-item active"
      >
        <span
          class="page-link"
        >
          1
          <span
            class="visually-hidden"
          >
            (current)
          </span>
        </span>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
          tabindex="0"
        >
          2
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
          tabindex="0"
        >
          3
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
          tabindex="0"
        >
          4
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
          tabindex="0"
        >
          5
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
          tabindex="0"
        >
          6
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
          tabindex="0"
        >
          7
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
          tabindex="0"
        >
          8
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
          tabindex="0"
        >
          9
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
          tabindex="0"
        >
          10
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
